rear_rider_device/rear_rider_bluetooth_server/src/main.py

```diff
@@ -0,0 +1,123 @@
+#!/usr/bin/env python3
+from sys import stdout
+from typing import Callable, Literal, Union
+
+from advertisement.rear_rider_adv import RearRiderAdvertisement
+
+from bluez.example_advertisement import LE_ADVERTISING_MANAGER_IFACE, TestAdvertisement
+from bluez.example_gatt_server import find_adapter, dbus, BLUEZ_SERVICE_NAME, GATT_MANAGER_IFACE
+from gi.repository import GLib
+from agent.simple import Agent
+
+from services.hello_world import HelloWorldService
+from rearrider_app import RearRiderApplication
+
+AGENT_MANAGER_IFACE = 'org.bluez.AgentManager1'
+AGENT_PATH = '/bluez/simpleagent'
+
+def main(print, on_ready: Union[None, Callable[[HelloWorldService], None]], on_read: Callable[[], str]):
```

Comment:
should we have multiple main functions? is there a reason one would be run directly over the other?

Reply:
The reason for this is because this file can be run as it's own program for testing purposes, but it can also be included as a library from another file.